File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res;
        int flag = 1;

        for(int j = 0; j < strs[0].size(); j++)
        {
            for(int i = 0; i < (strs.size() - 1); i++)
            {
                if(strs[i][j] == strs[i+1][j])
                {

                }
                else
                {
                    flag = 2;
                }
            }

            if(flag == 2)
            {
                break;
            }
            else
            {
                res += strs[0][j];
            }
        }
        
        return res;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);

        StringBuilder sb = new StringBuilder("");
        int minInd = -1;
        int n = strs.length;


        for(int i = 0; i < Math.min(strs[0].length(), strs[n-1].length()); i++)
        {
            if(strs[0].charAt(i) == strs[n-1].charAt(i))
            {
                minInd = i;
            }
            else
            {
                break;
            }
        }

        if(minInd == -1) return "";

        return strs[0].substring(0,minInd + 1);
    }
}